packages/ember-uploader/lib/file-field.js

```diff
@@ -1,13 +1,22 @@
+var deprecate = Ember.deprecate;
 var set = Ember.set;
+var on = Ember.on;
 
-export default Ember.TextField.extend({
+export default Ember.TextField.extend(Ember.Evented, {
   type: 'file',
   attributeBindings: ['multiple'],
   multiple: false,
   change: function(e) {
     var input = e.target;
     if (!Ember.isEmpty(input.files)) {
+      this.trigger('filesDidChange', input.files);
       set(this, 'files', input.files);
```

Comment:
I agree, a comment explaining this would be helpful. It can be removed in the next release.